tools/tdbg/app.go

```diff
@@ -0,0 +1,231 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tdbg
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"runtime/debug"
+	"strings"
+
+	"github.com/urfave/cli/v2"
+	"go.temporal.io/server/common/headers"
+
+	"github.com/temporalio/tctl-kit/pkg/color"
+	"github.com/temporalio/tctl/cli/dataconverter"
+	"github.com/temporalio/tctl/cli/headersprovider"
+	"github.com/temporalio/tctl/cli/plugin"
+)
+
+// SetFactory is used to set the ClientFactory global
+func SetFactory(factory ClientFactory) {
+	cFactory = factory
+}
+
+// NewCliApp instantiates a new instance of the CLI application.
+func NewCliApp() *cli.App {
+	app := cli.NewApp()
+	app.Name = "tdbg"
+	app.Usage = "A command-line tool for advanced debugging"
```

Comment:
I am on the outside of this PR, but I just want it to be known that to our users, often the term "debugging" relates to their code and its interaction with the SDK and is unrelated to the server. I just want to make sure there are no concerns here that we are taking the general term to be server-specific.

Reply:
renamed the Usage text to point to `Temporal server debugging` and still left the `debugging` terminology since as per slack conversation that's what this tool does and it lives in temporal server repo